clang-11 gives a number of warnings

Tried self-build clang-11 (very tip of clang master branch):
```
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:87:19: warning: anonymous non-C-compatible type given name for linkage purposes by typedef declaration; add a tag name here [-Wnon-c-typedef-for-linkage]
    typedef struct
                  ^
                   ExtVASurface
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:89:31: note: type is not C-compatible due to this default member initializer
        VASurfaceID surface = VA_INVALID_SURFACE;
                              ^~~~~~~~~~~~~~~~~~
/opt/intel/git/install/include/va/va.h:1275:28: note: expanded from macro 'VA_INVALID_SURFACE'
#define VA_INVALID_SURFACE      VA_INVALID_ID
                                ^~~~~~~~~~~~~
/opt/intel/git/install/include/va/va.h:1274:24: note: expanded from macro 'VA_INVALID_ID'
#define VA_INVALID_ID           0xffffffff
                                ^~~~~~~~~~
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:98:7: note: type is given name 'ExtVASurface' for linkage purposes by this typedef declaration
    } ExtVASurface;
      ^
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:87:19: warning: anonymous non-C-compatible type given name for linkage purposes by typedef declaration; add a tag name here [-Wnon-c-typedef-for-linkage]
    typedef struct
                  ^
                   ExtVASurface
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:89:31: note: type is not C-compatible due to this default member initializer
        VASurfaceID surface = VA_INVALID_SURFACE;
                              ^~~~~~~~~~~~~~~~~~
/opt/intel/git/install/include/va/va.h:1275:28: note: expanded from macro 'VA_INVALID_SURFACE'
#define VA_INVALID_SURFACE      VA_INVALID_ID
                                ^~~~~~~~~~~~~
/opt/intel/git/install/include/va/va.h:1274:24: note: expanded from macro 'VA_INVALID_ID'
#define VA_INVALID_ID           0xffffffff
                                ^~~~~~~~~~
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:98:7: note: type is given name 'ExtVASurface' for linkage purposes by this typedef declaration
    } ExtVASurface;
      ^
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:87:19: warning: anonymous non-C-compatible type given name for linkage purposes by typedef declaration; add a tag name here [-Wnon-c-typedef-for-linkage]
    typedef struct
                  ^
                   ExtVASurface
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:89:31: note: type is not C-compatible due to this default member initializer
        VASurfaceID surface = VA_INVALID_SURFACE;
                              ^~~~~~~~~~~~~~~~~~
/opt/intel/git/install/include/va/va.h:1275:28: note: expanded from macro 'VA_INVALID_SURFACE'
#define VA_INVALID_SURFACE      VA_INVALID_ID
                                ^~~~~~~~~~~~~
/opt/intel/git/install/include/va/va.h:1274:24: note: expanded from macro 'VA_INVALID_ID'
#define VA_INVALID_ID           0xffffffff
                                ^~~~~~~~~~
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:98:7: note: type is given name 'ExtVASurface' for linkage purposes by this typedef declaration
    } ExtVASurface;
      ^
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:87:19: warning: anonymous non-C-compatible type given name for linkage purposes by typedef declaration; add a tag name here [-Wnon-c-typedef-for-linkage]
    typedef struct
                  ^
                   ExtVASurface
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:89:31: note: type is not C-compatible due to this default member initializer
        VASurfaceID surface = VA_INVALID_SURFACE;
                              ^~~~~~~~~~~~~~~~~~
/opt/intel/git/install/include/va/va.h:1275:28: note: expanded from macro 'VA_INVALID_SURFACE'
#define VA_INVALID_SURFACE      VA_INVALID_ID
                                ^~~~~~~~~~~~~
/opt/intel/git/install/include/va/va.h:1274:24: note: expanded from macro 'VA_INVALID_ID'
#define VA_INVALID_ID           0xffffffff
                                ^~~~~~~~~~
/opt/intel/git/msdk/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h:98:7: note: type is given name 'ExtVASurface' for linkage purposes by this typedef declaration
    } ExtVASurface;
      ^


/opt/intel/git/msdk/_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp:6276:69: warning: bitwise or with non-zero value always evaluates to true [-Wtautological-bitwise-compare]
    if (true == executeParams.bSceneDetectionEnable && (FRC_ENABLED | config.m_extConfig.mode))
                                                        ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~

                                                        
/opt/intel/git/msdk/_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp:2810:9: warning: misleading indentation; statement is not part of the previous 'if' [-Wmisleading-indentation]
        if (((fo + 1) % gopPicSize == 0 && (gopOptFlag & MFX_GOP_CLOSED)) ||
        ^
/opt/intel/git/msdk/_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp:2806:5: note: previous statement is here
    if (fo % gopPicSize % gopRefDist == 0)
    ^


      
/opt/intel/git/msdk/_studio/mfx_lib/shared/src/mfx_vc1_dec_common.cpp:124:60: warning: overlapping comparisons always evaluate to true [-Wtautological-overlap-compare]
                if (mask != MFX_IOPATTERN_OUT_VIDEO_MEMORY || mask != MFX_IOPATTERN_OUT_SYSTEM_MEMORY || mask != MFX_IOPATTERN_OUT_OPAQUE_MEMORY)
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~



      
/opt/intel/git/msdk/_studio/mfx_lib/shared/src/mfx_vc1_dec_common.cpp:124:60: warning: overlapping comparisons always evaluate to true [-Wtautological-overlap-compare]
                if (mask != MFX_IOPATTERN_OUT_VIDEO_MEMORY || mask != MFX_IOPATTERN_OUT_SYSTEM_MEMORY || mask != MFX_IOPATTERN_OUT_OPAQUE_MEMORY)
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp
+++ b/_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp
@@ -2807,9 +2807,10 @@ mfxU8 GetFrameType(
         return (MFX_FRAMETYPE_P | MFX_FRAMETYPE_REF);
 
     //if ((gopOptFlag & MFX_GOP_STRICT) == 0)
-        if (((fo + 1) % gopPicSize == 0 && (gopOptFlag & MFX_GOP_CLOSED)) ||
-            (idrPicDist && (fo + 1) % idrPicDist == 0))
-            return (MFX_FRAMETYPE_P | MFX_FRAMETYPE_REF); // switch last B frame to P frame
+    if (((fo + 1) % gopPicSize == 0 && (gopOptFlag & MFX_GOP_CLOSED)) ||
+        (idrPicDist && (fo + 1) % idrPicDist == 0)) {
+        return (MFX_FRAMETYPE_P | MFX_FRAMETYPE_REF); // switch last B frame to P frame
+    }
 
 
     return (MFX_FRAMETYPE_B);

--- a/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h
+++ b/_studio/mfx_lib/shared/include/mfx_h264_encode_vaapi.h
@@ -1,4 +1,4 @@
-// Copyright (c) 2018-2019 Intel Corporation
+// Copyright (c) 2018-2020 Intel Corporation
 // 
 // Permission is hereby granted, free of charge, to any person obtaining a copy
 // of this software and associated documentation files (the "Software"), to deal
@@ -84,7 +84,7 @@ mfxStatus SetRoundingOffset(
 namespace MfxHwH264Encode
 {
     // map feedbackNumber <-> VASurface
-    typedef struct
+    struct ExtVASurface
     {
         VASurfaceID surface = VA_INVALID_SURFACE;
         mfxU32 number       = 0;
@@ -95,7 +95,7 @@ namespace MfxHwH264Encode
         VASurfaceID mbstat  = VA_INVALID_ID;
         VASurfaceID mbcode  = VA_INVALID_ID;
 #endif
-    } ExtVASurface;
+    };
 
     void UpdatePPS(
         DdiTask const & task,

--- a/_studio/mfx_lib/shared/src/mfx_vc1_dec_common.cpp
+++ b/_studio/mfx_lib/shared/src/mfx_vc1_dec_common.cpp
@@ -1,4 +1,4 @@
-// Copyright (c) 2017-2019 Intel Corporation
+// Copyright (c) 2017-2020 Intel Corporation
 // 
 // Permission is hereby granted, free of charge, to any person obtaining a copy
 // of this software and associated documentation files (the "Software"), to deal
@@ -121,8 +121,11 @@ mfxStatus MFXVC1DecCommon::Query(VideoCORE* core, mfxVideoParam *in, mfxVideoPar
                 (in->IOPattern & MFX_IOPATTERN_OUT_OPAQUE_MEMORY))
             {
                 uint32_t mask = in->IOPattern & 0xf0;
-                if (mask != MFX_IOPATTERN_OUT_VIDEO_MEMORY || mask != MFX_IOPATTERN_OUT_SYSTEM_MEMORY || mask != MFX_IOPATTERN_OUT_OPAQUE_MEMORY)
+                if ((mask == MFX_IOPATTERN_OUT_SYSTEM_MEMORY) || (mask == MFX_IOPATTERN_OUT_VIDEO_MEMORY) ||
+                    (mask == MFX_IOPATTERN_OUT_OPAQUE_MEMORY))
                     out->IOPattern = in->IOPattern;
+                else
+                    sts = MFX_ERR_UNSUPPORTED;
             }
 
             if (in->Protected)

--- a/_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp
+++ b/_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp
@@ -128,7 +128,6 @@ static void MemSetZero4mfxExecuteParams (mfxExecuteParams *pMfxExecuteParams )
     pMfxExecuteParams->Contrast = 0;
     pMfxExecuteParams->Hue = 0;
     pMfxExecuteParams->Saturation = 0;
-    pMfxExecuteParams->bSceneDetectionEnable = false;
     pMfxExecuteParams->bVarianceEnable = false;
     pMfxExecuteParams->bImgStabilizationEnable = false;
     pMfxExecuteParams->istabMode = 0;
@@ -6178,7 +6177,6 @@ mfxStatus ConfigureExecuteParams(
                 }
                 else if (MFX_EXTBUFF_VPP_SCENE_ANALYSIS == bufferId)
                 {
-                    executeParams.bSceneDetectionEnable = false;
                 }
                 else if (MFX_EXTBUFF_VPP_DETAIL == bufferId)
                 {
@@ -6270,13 +6268,6 @@ mfxStatus ConfigureExecuteParams(
         {
             config.m_extConfig.mode = FRC_DISABLED | FRC_DISTRIBUTED_TIMESTAMP;
         }
-    }
-
-
-    if (true == executeParams.bSceneDetectionEnable && (FRC_ENABLED | config.m_extConfig.mode))
-    {
-        // disable scene detection
-        executeParams.bSceneDetectionEnable = false;
     }
 
     if ( (0 == memcmp(&videoParam.vpp.In, &videoParam.vpp.Out, sizeof(mfxFrameInfo))) &&

--- a/_studio/mfx_lib/vpp/src/mfx_vpp_jpeg_d3d9.cpp
+++ b/_studio/mfx_lib/vpp/src/mfx_vpp_jpeg_d3d9.cpp
@@ -1,4 +1,4 @@
-// Copyright (c) 2017 Intel Corporation
+// Copyright (c) 2017-2020 Intel Corporation
 // 
 // Permission is hereby granted, free of charge, to any person obtaining a copy
 // of this software and associated documentation files (the "Software"), to deal
@@ -55,7 +55,6 @@ static void MemSetZero4mfxExecuteParams (mfxExecuteParams *pMfxExecuteParams )
     pMfxExecuteParams->Contrast = 0;
     pMfxExecuteParams->Hue = 0;
     pMfxExecuteParams->Saturation = 0;
-    pMfxExecuteParams->bSceneDetectionEnable = false;
     pMfxExecuteParams->bVarianceEnable = false;
     pMfxExecuteParams->bImgStabilizationEnable = false;
     pMfxExecuteParams->istabMode = 0;

--- a/_studio/shared/include/mfx_vpp_interface.h
+++ b/_studio/shared/include/mfx_vpp_interface.h
@@ -275,7 +275,6 @@ namespace MfxHwVideoProcessing
                ,Contrast(0)
                ,Hue(0)
                ,Saturation(0)
-               ,bSceneDetectionEnable(false)
                ,bVarianceEnable(false)
                ,bImgStabilizationEnable(false)
                ,istabMode(0)
@@ -342,7 +341,6 @@ namespace MfxHwVideoProcessing
                     detailFactor != 0 ||
                     iTargetInterlacingMode != 0 ||
                     bEnableProcAmp != false ||
-                    bSceneDetectionEnable != false ||
                     bVarianceEnable != false ||
                     bImgStabilizationEnable != false ||
                     bFRCEnable != false ||
@@ -397,8 +395,6 @@ namespace MfxHwVideoProcessing
         mfxF64         Contrast;
         mfxF64         Hue;
         mfxF64         Saturation;
-
-        bool           bSceneDetectionEnable;
 
         bool           bVarianceEnable;
         bool           bImgStabilizationEnable;

